app/models/vm_or_template.rb

```diff
@@ -472,6 +472,64 @@ def self.invoke_task_local(task, vm, options, args)
     )
   end
 
+  def self.invoke_tasks_remote_queue(options, deliver_on = nil)
+    queue_opts = {
+      :role        => "user_interface",
+      :class_name  => base_class.name,
+      :method_name => 'invoke_tasks_remote',
+      :args        => [options]
+    }
+    queue_opts[:deliver_on] = deliver_on if deliver_on
+    MiqQueue.put(queue_opts)
+  end
+
+  def self.invoke_api_task(hostname, action, ids)
+    require 'rest-client'
+    require 'json'
+
+    api_user     = Settings.webservices.remote_miq_api.user
+    api_password = MiqPassword.try_decrypt(Settings.webservices.remote_miq_api.password)
+
+    raise "No API username provided for remote #{action}" if api_user.nil?
+    raise "No API password provided for user #{api_user} for remote #{action}" if api_password.nil?
+
+    uri = URI::HTTPS.build(
+      :host     => hostname,
+      :userinfo => "#{api_user}:#{api_password}",
+      :path     => "/api/vms"
+    )
+
+    request = {
+      "action"    => action,
+      "resources" => ids.collect { |id| {"id" => id} }
+    }
+
+    begin
+      timeout = Settings.webservices.remote_miq_api.request_timeout || 600
```

Comment:
The `|| 600` should not be necessary if you just hardcode the value in the settings